Convert maintainer file entries to attributes, add github handles

Profpatsch opened the pull request:
Based on https://github.com/NixOS/nixpkgs/pull/34842, the
nix-instantiate output was pretty-printed and the validity of the github handles
manually verified, by automatically checking whether the user handles exist on
github (https://github.com/userhandle, status 200 or 404).
Each handle under 5 characters was manually checked (because the collision
probability with non-maintainer accounts is high), each missing entry was
manually researched.

The script used is kept in `maintainers/scripts` as an example of how to work
with the mainainers list through nix’ JSON interface.

Hydra might need to be patched.

cc @FRidh @vcunat

Profpatsch committed:
Convert maintainer file entries to attributes, add github handles

Based on https://github.com/NixOS/nixpkgs/pull/34842, the
nix-instantiate output was pretty-printed and the validity of the github handles
manually verified, by automatically checking whether the user handles exist on
github (https://github.com/userhandle, status 200 or 404).
Each handle under 5 characters was manually checked (because the collision
probability with non-maintainer accounts is high), each missing entry was
manually researched.

The script used is kept in `maintainers/scripts` as an example of how to work
with the mainainers list through nix’ JSON interface.

FRidh commented:
Thanks for the effort. Note the merge conflict though.

cillianderoiste commented on lib/maintainers.nix:
```diff
@@ -1,799 +1,3946 @@
+  };
+  dizfer = {
+    email = "david@izquierdofernandez.com";
+    githu
```
My github id is cillianderoiste, thanks!

Profpatsch replied:
Thanks, fixed.

Profpatsch commented:
> Note the merge conflict though.

Yeah, I expect this to take a while to merge, so there might be multiple changes to maintainers nix by then.

FRidh commented:
Right. That's why I suggested merging https://github.com/NixOS/nixpkgs/pull/34842 first, then fixing Hydra, and when that is done convert to what you have here.

jtojnar commented on lib/maintainers.nix:
```diff
@@ -1,799 +1,3946 @@
+  };
+  dizfer = {
+    email = "david@izquierdofernandez.com";
+    githu
```
Should be @gavinrogers, see https://github.com/NixOS/nixpkgs/pull/35714

Profpatsch replied:
Something is broken with these markers, every one points to `dizfer` somehow?

jtojnar replied:
`gavin` → `gavinrogers`

jtojnar commented on lib/maintainers.nix:
```diff
@@ -1,799 +1,3946 @@
+  };
+  dizfer = {
+    email = "david@izquierdofernandez.com";
+    githu
```
@bignaux

jtojnar replied:
`genesis` → `bignaux`

bignaux replied:
ok

jtojnar commented on lib/maintainers.nix:
```diff
@@ -1,799 +1,3946 @@
+  };
+  dizfer = {
+    email = "david@izquierdofernandez.com";
+    githu
```
@edolstra

jtojnar replied:
`eelco` → `edolstra

Profpatsch commented:
It looks like cross-checking entries <5 chars is not enough, normal dictionary words will most likely also overlap.

Profpatsch commented:
> That's why I suggested merging #34842 first

The problem with that commit is that lots of github handles are simply wrong, because the maintainer names don’t translate. So we are going to mention lots of non-nixos people.

jtojnar commented:
How hard would it be to check if given handle contributed to the nixpkgs repo and (manually) fix those that did not?

7c6f434c commented:
Well, may vanity counter does build a correspondence of names, emails and github names

7c6f434c commented on lib/maintainers.nix:
```diff
@@ -1,799 +1,3946 @@
+  };
+  dizfer = {
+    email = "david@izquierdofernandez.com";
+    githu
```
That's actually me, 7c6f434c

jtojnar replied:
`raskin` → `7c6f434c`

jtojnar replied:
GitHub is really weird today.

Profpatsch commented:
> How hard would it be to check if given handle contributed to the nixpkgs repo and (manually) fix those that did not?

Probably not very, might be sensible to use the Github API for that and rewrite the script (in the commit) a bit.

jtojnar commented:
`curl "https://github.com/NixOS/nixpkgs/commits?author=$HANDLE" | grep 'No commits found'` should be good enough.

Profpatsch committed:
maintainers/scripts: improve github handle checker

Now also finds name-clashes for github handles who never contributed to nixpkgs
before. Also deals with too many request errors.

Profpatsch commented:
Checking for commits right now, this finds quite a few accounts that are not commiters. `curl`ing the github repo is pretty slow, since github has very hard limits on the amount of requests possible (and returns HTTP `429` in that case).

Profpatsch committed:
lib/maintainers: correct wrong (clashing) github handles

Corrected every handle that had no commits to nixpkgs, manually researched the
correct handles by looking at maintained packages & blames/history on Github.

Profpatsch commented:
Okay, here we go. Now every handle that had no nixpkgs commits is corrected!

Profpatsch commented:
Thanks to @aszlig’s input, I added a `shortName` property, which is used by hydra to send mails. @FRidh this should supersede #34842, minus the need for parsing.

Profpatsch commented:
Removed the `shortName` again, @grahamc said we should fix hydra instead. According to @aszlig’s and my research, this will only lead to hydra not sending any mails for the time being.

oxij commented:
Yay, nice!

Can we also rename this file to `maintainers/maintainers-list.nix` while we at this? Motivation: much saner `git log ./lib`. I wanted to do it for a while now, but was too lazy to make a separate PR for that.

jtojnar commented on lib/maintainers.nix:
```diff
@@ -1319,7 +1313,6 @@
   };
   fuzzy-id = {
     email = "hacking+nixos@babibo.de";
-    github = "fuzzy-id";
```
https://github.com/NixOS/nixpkgs/pull/29147

jtojnar commented on lib/maintainers.nix:
```diff
@@ -1150,7 +1145,6 @@
   };
   epitrochoid = {
     email = "mpcervin@uncg.edu";
-    github = "epitrochoid";
```
https://github.com/NixOS/nixpkgs/pull/9243

jtojnar commented on lib/maintainers.nix:
```diff
@@ -1020,7 +1017,6 @@
   };
   drewkett = {
     email = "burkett.andrew@gmail.com";
-    github = "drewkett";
```
https://github.com/NixOS/nixpkgs/pull/14878

jtojnar reviewed (changes requested):
Maybe also run `curl "https://github.com/NixOS/nixpkgs/pulls?utf8=%E2%9C%93&q=author%3A${user}" | grep 'No results matched your search'` for users that commit without matching e-mail address.

Profpatsch committed:
lib/maintainers: add new maintainers before maintainers merge

Add new maintainers that were added before merging the new `maintainers.nix`
file format.

Profpatsch commented:
@jtojnar I don’t understand, why is this a problem?

Profpatsch committed:
lib/maintainers: rename file to maintainers-list.nix

Based on a request by @oxij:
“Can we also rename this file to `maintainers/maintainers-list.nix` while we at
this? Motivation: much saner `git log ./lib`.”

Profpatsch commented:
Okay, since I brought it manually up-to-date with master, I’m going to merge now.
@jtojnar let’s put further discussion/changes into issues/more pull requests.

Don’t hesitate to revert/amend should this change break something important.

oxij commented:
See #36275.

Profpatsch commented:
@GrahamcOfBorg eval

jtojnar commented:
@Profpatsch the problem is, you are removing GitHub handles that have commits in nixpkgs (but GitHub cannot pair them).

Profpatsch commented:
I removed every handle where Github couldn’t resolve to a user page, yes, since those are either commits from non-Github users (via normal git commits) or Github accounts that don’t exist anymore.

jtojnar commented:
At least the three I mentioned do resolve to a user page, are commits by GitHub users, and the accounts do still exist, only GitHub cannot pair them correctly (due to e-mail address mismatch). Checking the pull requests should fix these unpaired accounts.

Then only renamed accounts will remain. They need to be checked manually by checking the git log for expressions by the maintainer.

Profpatsch commented:
Ah, I see. Can you open a new issue (or maybe even pull request) for those cases? Thanks for the scrutiny!
